geomstats/learning/riemannian_mean_shift.py

```diff
@@ -0,0 +1,161 @@
+"""
+Riemannian mean-shift clustering
+Lead author: Nina Miolane and Shubham Talbar
+"""
+
+import joblib
+from sklearn.base import BaseEstimator, ClusterMixin
+
+import geomstats.backend as gs
+from geomstats.learning.frechet_mean import FrechetMean
+
+
+class RiemannianMeanShift(ClusterMixin, BaseEstimator):
+    """
+    Class for Riemannian Mean Shift algorithm on manifolds.
+    Mean Shift is a procedure for locating the maxima - the modes of a
+    density function given discrete data sampled from that function. It is
+    an iterative method for finding the centers of a collection of clusters.
+
+    Following implementation assumes a flat kernel method.
+
+    Parameters
+    ----------
+    manifold : object of class RiemannianManifold
+        Geomstats Riemannian manifold on which the
+        Riemannian Mean Shift algorithm is to applied.
+    metric : object of class RiemannianMetric
+        Geomstats Riemannian metric associated to the space used.
+    bandwidth : size of neighbourhood around each center
+        All points in 'bandwidth' size around center are considered for
+        calculating new mean centers.
+    tot : float, stopping condition
+        Computation of subsequent mean centers is stopped when the distance
+        between them is less than 'tot'
+    mean : Mean method used to calculate the new center of a cluster
+    n_centers : Total initial centers
+    n_jobs : Number of parallel threads to be initiated for parallel computation
+    max_iter : int
+        Upper bound on total number of iterations for the centers to converge
+    init_centers : Initializing centers, either from the given input points or
+             random points uniformly distributed in the input manifold
+    kernel : Weighing function to assign kernel weights to each center
+    """
+
+    def __init__(
+        self,
+        manifold,
+        metric,
+        bandwidth,
+        tol=1e-2,
+        n_centers=1,
+        n_jobs=1,
+        max_iter=100,
+        init_centers="from_points",
+        kernel="flat",
+        **frechet_mean_kwargs
+    ):
+        self.manifold = manifold
+        self.metric = metric
+        self.bandwidth = bandwidth
+        self.tol = tol
+        self.mean = FrechetMean(self.metric, **frechet_mean_kwargs)
+        self.n_centers = n_centers
+        self.n_jobs = n_jobs
+        self.max_iter = max_iter
+        self.init_centers = init_centers
+        self.kernel = kernel
+        self.centers = None
+
+        print(type(max_iter))
+        print(max_iter)
+
+    def __intersets_distances(self, points_A, points_B, **joblib_kwargs):
```

Comment:
This function is great! Can't believe we did not have it (@nguigs ). Could rename it to `dist_intersets` and put it on the Line 389 on this file?
https://github.com/geomstats/geomstats/blob/master/geomstats/geometry/riemannian_metric.py#L389

It would come just after:
- dist_broadcast
- dist_pairwise
which are doing similar computations (but not quite).

It would also become a public function, which means that a docstring will be required (with  parameters' description).